### PR TITLE
[#4295] feat(server): Add REST API for the owner

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add REST API for the owner.

### Why are the changes needed?

Fix: #4295 

### Does this PR introduce _any_ user-facing change?
I will add the document in the later pull request.

### How was this patch tested?
Add the new ut.

### CODE REVIEW

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import org.apache.gravitino.dto.util.DTOConverters;
+import org.apache.gravitino.metrics.MetricNames;
+import org.apache.gravitino.server.authorization.NameBindings;
+import org.apache.gravitino.server.web.Utils;
+
+@NameBindings.AccessControlInterfaces
+@Path("/metalakes/{metalake}/owners")
+public class OwnerOperations {
+
+  private final OwnerManager ownershipManager;
```
Comment (xunliu) [inline server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:51]: Please change to `ownerManager`.

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+          MetadataObjects.parse(
+              fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            Optional<Owner> owner = ownershipManager.getOwner(metalake, object);
+            if (owner.isPresent()) {
+              return Utils.ok(new OwnerResponse(DTOConverters.toDTO(owner.get())));
+            } else {
+              return Utils.ok(new MessageResponse("Owner is not set"));
```
Comment (xunliu) [inline server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:82]: Whether an empty `OwnerResponse("")` can be return?
Additionally, A RESTful interface can return two different type (Owner|Message) Responses, I think this uncommon, and will increase the burden of the interface caller.

File: server/src/test/java/org/apache/gravitino/server/web/rest/TestOwnerOperations.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+          @Override
+          public Type type() {
+            return Type.USER;
+          }
+        };
+
+    when(manager.getOwner(any(), any())).thenReturn(Optional.of(owner));
+
+    Response resp =
+        target("/metalakes/metalake1/owners/metalake/metalake1")
```
Comment (xunliu) [inline server/src/test/java/org/apache/gravitino/server/web/rest/TestOwnerOperations.java:122]: I think this request path is very confusing.
If we need process schema, then the request path is `/metalakes/metalake1/owners/metalake/metalake1/catalog1/schema1` ?
Cloud there be a better design?
`/metalake1/catalog1/schema1/owner` ?

Comment (jerqi) [reply server/src/test/java/org/apache/gravitino/server/web/rest/TestOwnerOperations.java:122]: /metalakes/metalake1/owners/schema/metalake1.catalog1.schema1. This style is the same as the tag's.

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java (hunk 1 above)
Comment (jerqi) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:51]: OK.

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java (hunk 2 above)
Comment (jerqi) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java]: I make owner is nullable and delete MessageResponse.

File: server/src/test/java/org/apache/gravitino/server/web/rest/TestOwnerOperations.java (hunk 3 above)
Comment (xunliu) [reply server/src/test/java/org/apache/gravitino/server/web/rest/TestOwnerOperations.java:121]: OK

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java (hunk 2 above)
Comment (xunliu) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java]: OK

**Comment by xunliu** [comment]
@yuqi1129 Please help review this PR, thanks

**Comment by yuqi1129** [comment]
> @yuqi1129 Please help review this PR, thanks

Got it

File: common/src/main/java/org/apache/gravitino/dto/authorization/OwnerDTO.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.dto.authorization;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.gravitino.authorization.Owner;
+
+/** Represents a Owner Data Transfer Object (DTO). */
```
Comment (yuqi1129) [inline common/src/main/java/org/apache/gravitino/dto/authorization/OwnerDTO.java:24]: a -> an

File: common/src/main/java/org/apache/gravitino/dto/responses/OwnerResponse.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    this.owner = null;
+  }
+
+  @Override
+  public void validate() throws IllegalArgumentException {
+    super.validate();
+
+    if (owner != null) {
+      Preconditions.checkArgument(
+          StringUtils.isNotBlank(owner.name()), "owner 'name' must not be null and empty");
```
Comment (yuqi1129) [inline common/src/main/java/org/apache/gravitino/dto/responses/OwnerResponse.java:62]: not be null or empty

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  @PUT
+  @Path("{type}/{fullName}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "set-object-owner." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "set-object-owner", absolute = true)
+  public Response setOwnerForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      OwnerSetRequest request) {
```
Comment (yuqi1129) [inline server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:99]: What's the difference between `type` in  `OwnerSetRequest`  and type in line 97.

File: common/src/main/java/org/apache/gravitino/dto/authorization/OwnerDTO.java (hunk 4 above)
Comment (jerqi) [reply common/src/main/java/org/apache/gravitino/dto/authorization/OwnerDTO.java:24]: Done.

File: common/src/main/java/org/apache/gravitino/dto/responses/OwnerResponse.java (hunk 5 above)
Comment (jerqi) [reply common/src/main/java/org/apache/gravitino/dto/responses/OwnerResponse.java:62]: OK.

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java (hunk 6 above)
Comment (jerqi) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:99]: One is the type of owner, the other is the type of metadata object.

Comment (yuqi1129) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:99]: Can you rename the parameter to `owerType` or something similar?

Comment (jerqi) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:99]: I hesitate to add this. I worry whether it is clear enough. The type of `type` is Owner.Type. Is it enough?

Comment (jerqi) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java:99]: Change 97Line type to metadataObjectType.

File: server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  @GET
+  @Path("{type}/{fullName}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-object-owner." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-object-owner", absolute = true)
+  public Response getOwnerForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
```
[the remark this answers is not in the document]
Comment (jerqi) [reply server/src/main/java/org/apache/gravitino/server/web/rest/OwnerOperations.java]: Fixed.

**Review by xunliu — COMMENTED** [review]
@jerryshao Please help review this PR, Thanks.

**Review by xunliu — APPROVED** [review]
If no more comment, I will merge this PR.

**Pull request merged**